tests/sentry/utils/sdk_crashes/test_sdk_crash_detection.py

```diff
@@ -0,0 +1,335 @@
+                    {
+                        "function": "std::__terminate",
+                        "symbol": "_ZSt11__terminatePFvvE",
+                        "package": "libc++abi.dylib",
+                        "in_app": False,
+                    },
+                ]
+            ),
+            True,
+            mock_sdk_crash_reporter,
+        )
+
+    def test_frames_only_in_app_after_sentry_frame_not_reported(self, mock_sdk_crash_reporter):
+        self.execute_test(
+            get_crash_event_with_frames(
+                [
+                    {
+                        "function": "std::__terminate",
+                        "symbol": "_ZSt11__terminatePFvvE",
+                        "package": "libc++abi.dylib",
+                        "in_app": False,
+                    },
+                    get_sentry_frame("sentrycrashdl_getBinaryImage"),
+                    {
+                        "function": "_objc_terminate",
+                        "symbol": "_ZL15_objc_terminatev",
+                        "package": "libobjc.A.dylib",
+                        "in_app": False,
+                    },
+                    {
+                        "function": "__handleUncaughtException",
+                        "symbol": "__handleUncaughtException",
+                        "package": "CoreFoundation",
+                        "in_app": False,
+                    },
+                    IN_APP_FRAME,
+                ]
+            ),
+            False,
+            mock_sdk_crash_reporter,
+        )
+
+
+class SDKCrashReportTestMixin(BaseSDKCrashDetectionMixin, SnubaTestCase):
+    @pytest.mark.django_db
+    def test_sdk_crash_event_stored_to_sdk_crash_project(self):
+
+        cocoa_sdk_crashes_project = self.create_project(
+            name="Cocoa SDK Crashes",
+            slug="cocoa-sdk-crashes",
+            teams=[self.team],
+            fire_project_created=True,
+        )
+
+        event = self.create_event(
+            data=get_crash_event(),
+            project_id=self.project.id,
+        )
+
+        with override_settings(SDK_CRASH_DETECTION_PROJECT_ID=cocoa_sdk_crashes_project.id):
+            sdk_crash_event = sdk_crash_detection.detect_sdk_crash(event=event)
+
+            assert sdk_crash_event is not None
+
+            event_store = SnubaEventStorage()
+            fetched_sdk_crash_event = event_store.get_event_by_id(
+                cocoa_sdk_crashes_project.id, sdk_crash_event.event_id
+            )
+
+            assert cocoa_sdk_crashes_project.id == fetched_sdk_crash_event.project_id
+            assert sdk_crash_event.event_id == fetched_sdk_crash_event.event_id
+
+
+@region_silo_test
+class SDKCrashDetectionTest(
+    TestCase,
+    CococaSDKTestMixin,
+    CococaSDKFunctionTestMixin,
+    CococaSDKFilenameTestMixin,
+    CococaSDKFramesTestMixin,
+    PerformanceEventTestMixin,
+    SDKCrashReportTestMixin,
+):
```

Comment:
please avoid writing new tests as class tests

Reply:
Thanks for pointing that out. I already created an issue for investigating which tests I can convert to function tests: https://github.com/getsentry/sentry/issues/50061